synapse/storage/databases/main/events.py

```diff
@@ -1770,10 +1770,21 @@ def _handle_insertion_event(self, txn: LoggingTransaction, event: EventBase):
             # Not a insertion event
             return
 
-        # Skip processing a insertion event if the room version doesn't
-        # support it.
+        # Skip processing an insertion event if the room version doesn't
+        # support it or the event is not from the room creator.
         room_version = self.store.get_room_version_txn(txn, event.room_id)
-        if not room_version.msc2716_historical:
+        room_creator = self.db_pool.simple_select_one_onecol_txn(
+            txn,
+            table="rooms",
+            keyvalues={"room_id": event.room_id},
+            retcol="creator",
+            allow_none=True,
+        )
+        if (
+            not room_version.msc2716_historical
+            or not self.hs.config.experimental.msc2716_enabled
```

Comment:
I think I am going to have to add some new background updates to fill in the `insertion_events`, `insertion_event_edges`, `chunk_events`, and `insertion_event_extremities` tables when `hs.config.experimental.msc2716_enabled` is enabled so homeservers which go from having the config disabled to enabled(or unsupported to supported) can fill in the necessary info to have historical events appear. Am I thinking about this correctly?

Reply:
Yes probably. If you don't add a background job then it'll only work for servers that were upgraded to a suitable version at the time they processed the events, which is fine for testing but probably something we want to fix before using it for real.

Reply:
Tracked in https://github.com/matrix-org/synapse/issues/10737